master: add info about tidb_opt_enable_alternative_logical_plans (#23218)
pingcap/docs: correlated-subquery-optimization.md

Therefore, when there are few external values, do not perform decorrelation, which might bring better execution performance. In this case, you can disable this optimization by using the [`NO_DECORRELATE`](/optimizer-hints.md#no_decorrelate) optimizer hint or by disabling the "subquery decorrelation" optimization rule in the [blocklist of optimization rules and expression pushdown](/blocklist-control-plan.md). In most cases, it is recommended to use the optimizer hint along with [SQL Plan Management](/sql-plan-management.md) to disable the decorrelation.

> Starting from TiDB v8.5.7 and v9.0.0, you can also use the system variable [`tidb_opt_enable_alternative_logical_plans`](/system-variables.md#tidb_opt_enable_alternative_logical_plans-new-in-v857-and-v900) to optimize such scenarios. After you enable this variable, if the candidate plan after [correlated subquery decorrelation](/correlated-subquery-optimization.md) does not generate an equivalent `IndexJoin` candidate with the same access direction as the original correlated subquery, the optimizer additionally keeps a "non-decorrelated" candidate plan, evaluates both "decorrelated" and "non-decorrelated" candidate plans, and finally selects the [execution plan](/explain-subqueries.md) with the lower cost.

Author: ```suggestion
Starting from v8.5.7 and v9.0.0, you can also use the system variable [`tidb_opt_enable_alternative_logical_plans`](/system-variables.md#tidb_opt_enable_alternative_logical_plans-new-in-v857-and-v900) to optimize such scenarios. After you enable this variable, if the decorrelated candidate plan fails to generate an equivalent `IndexJoin` candidate plan with the same access direction as the original correlated subquery, the optimizer additionally retains a non-decorrelated candidate plan, evaluates both the decorrelated and non-decorrelated candidate plans, and selects the [execution plan](/explain-subqueries.md) with the lower cost.
```
